pallets/xst/src/lib.rs

```diff
@@ -1087,6 +1090,58 @@ impl<T: Config> LiquiditySource<T::DEXId, T::AccountId, T::AssetId, Balance, Dis
         )
     }
 
+    fn step_quote(
+        dex_id: &T::DEXId,
+        input_asset_id: &T::AssetId,
+        output_asset_id: &T::AssetId,
+        amount: QuoteAmount<Balance>,
+        samples_count: usize,
+        deduce_fee: bool,
+    ) -> Result<VecDeque<SwapChunk<Balance>>, DispatchError> {
+        if !Self::can_exchange(dex_id, input_asset_id, output_asset_id) {
+            fail!(Error::<T>::CantExchange);
+        }
+        if amount.amount().is_zero() {
+            return Ok(VecDeque::new());
+        }
+
+        let synthetic_base_asset_id = &T::GetSyntheticBaseAssetId::get();
+        let (input_amount, output_amount, fee_amount) = if input_asset_id == synthetic_base_asset_id
+        {
+            Self::decide_sell_amounts(&input_asset_id, &output_asset_id, amount, deduce_fee, false)?
+        } else {
+            Self::decide_buy_amounts(&output_asset_id, &input_asset_id, amount, deduce_fee, false)?
+        };
+
+        let mut monolith = SwapChunk::new(input_amount, output_amount, fee_amount);
+
+        let limit = T::GetSyntheticBaseBuySellLimit::get();
+
+        if input_asset_id == synthetic_base_asset_id {
+            if input_amount > limit {
+                monolith = monolith
+                    .rescale_by_input(limit)
+                    .ok_or(Error::<T>::PriceCalculationFailed)?;
+            }
+        } else {
+            if output_amount > limit {
+                monolith = monolith
+                    .rescale_by_output(limit)
+                    .ok_or(Error::<T>::PriceCalculationFailed)?;
+            }
+        }
+
+        let ratio = (FixedWrapper::from(1) / FixedWrapper::from(samples_count))
+            .get()
+            .map_err(|_| Error::<T>::PriceCalculationFailed)?;
+
+        let chunk = monolith
+            .rescale_by_ratio(ratio)
+            .ok_or(Error::<T>::PriceCalculationFailed)?;
+
+        Ok(vec![chunk; samples_count].into())
```

Comment:
isn't it possible to use `inner_quote` since no splitting into chunks occurs?

Reply:
maybe I miss something, did not check thoroughly

Reply:
it is split, it returns vector of equal chunks

Reply:
Missed it, ok)